### PR TITLE
Avoid compiler crash when parsing standalone `[]`

### PR DESCRIPTION
**Pull request by vasslitvinov**
With this change, issue a proper parser error upon `type T = [];`
whereas formerly it crashed the compiler.

### CODE REVIEW

File: compiler/AST/build.cpp [hunk 1]
```diff
@@ -1035,6 +1035,9 @@ buildForallLoopExpr(Expr* indices, Expr* iteratorExpr, Expr* expr, Expr* cond, b
 //
 Expr* buildForallLoopExprFromArrayType(CallExpr* buildArrTypeCall,
                                            bool recursiveCall) {
+  if (buildArrTypeCall->isPrimitive(PRIM_ERROR))  // ex. 'type T = [];'
+    return buildArrTypeCall;
+
```
Comment (lydia-duncan) [inline compiler/AST/build.cpp:1040]: Might be good to put an INT_ASSERT that it's not another type of primitive, since I'm assuming that would also crash the compiler on line 1042

File: test/arrays/compilerErrors/array-generic-type.chpl [hunk 2]
```diff
@@ -0,0 +1,7 @@
+// Should this be allowed?
+// For now, generate a parser error.
+
+type T = [];
+
+proc test(type arg) { }
+test([]);
```
Comment (lydia-duncan) [inline test/arrays/compilerErrors/array-generic-type.chpl:7]: I'd also have a `test(T);` line here, just in case we ever do enable it

File: test/functions/typeMethods/arg-type-type-method.bad [hunk 3]
```diff
@@ -1,2 +1,2 @@
-arg-type-type-method.chpl:6: In function 'test':
```
Comment (lydia-duncan) [inline test/functions/typeMethods/arg-type-type-method.bad:1]: It'd be nice to do this in a separate PR, but it's good that you call it out in the PR message.  Should it also be mentioned in the main subject line?  I think you have space to do so

**Review by lydia-duncan — COMMENTED** [review]
I'd like to have the changes I suggest here made, but I don't need to take a look again unless there's a discussion about them.

**Comment by lydia-duncan** [comment]
Glad it was so simple to fix!

File: compiler/AST/build.cpp (hunk 1 above)
Comment (vasslitvinov) [reply compiler/AST/build.cpp:1040]: Turns out Line 1042 is guarded against `baseExpr` being a primitive. So no change is needed there.

Comment (lydia-duncan) [reply compiler/AST/build.cpp:1040]: How so?  Is the PRIM_ERROR case on line 1038 the only one that would be allowed into this function?

Comment (vasslitvinov) [reply compiler/AST/build.cpp:1040]: Oops, I looked at the wrong line.

The line `UnresolvedSymExpr* ursym = toUnresolvedSymExpr(buildArrTypeCall->baseExpr);` will result in `ursym` being NULL if `buildArrTypeCall` is a primitive.
